controllers/complianceeventsapi/types.go

```diff
@@ -236,3 +385,130 @@ func (j *JSONMap) Scan(src interface{}) error {
 
 	return json.Unmarshal(source, j)
 }
+
+type dbField struct {
+	// value is the underlying value set on the struct field.
+	value any
+	// fieldIndex is the index used to access this struct field on the struct instance this field is a part of. This is
+	// useful to set the value on the struct after a database query.
+	fieldIndex int
+	// goType is the Go type of this struct field. This is used to marhsall a SELECT query result to a struct.
+	goType reflect.Type
+}
+
+// getFields parses the input object and returns a map where the keys are database column names and the values are
+// dbField instances representing the struct fields on the input object.
+func getFields(obj any) map[string]dbField {
+	fields := map[string]dbField{}
+
+	values := reflect.Indirect(reflect.ValueOf(obj))
+	typesOf := values.Type()
+
+	for i := 0; i < values.NumField(); i++ {
+		structField := typesOf.Field(i)
+
+		fieldName := structField.Tag.Get("db")
+		if fieldName == "" {
+			// Skip struct fields that are not database columns.
+			continue
+		}
+
+		reflectField := values.Field(i)
+
+		field := dbField{
+			fieldIndex: i,
+			goType:     structField.Type,
+		}
+
+		if reflectField.IsZero() {
+			field.value = nil
+		} else {
+			field.value = reflect.Indirect(reflectField).Interface()
+		}
+
+		fields[fieldName] = field
+	}
+
+	return fields
+}
+
+// fromRow assigns the output from `rows.Scan` to the input `obj`. Note that this does not explicitly support
+// nested structs.
+func fromRow(rows *sql.Rows, obj any) error {
```

Comment:
Especially if you slimmed the `dbRow` interface, we could rename this function to `ScanInto`. To me, it feels slightly more representative of what's going on.

Reply:
When writing the GET HTTP endpoint, I found myself not using this because I always had a single large query with several joins and just referenced the fields directly in a single scan, so I think I'll just remove this.